interfaces/IMediaPlayer.h

```diff
@@ -28,27 +28,34 @@ struct IMediaPlayer : virtual public Core::IUnknown {
 
     enum { ID = ID_MEDIAPLAYER };
 
-    struct INotification : virtual public Core::IUnknown {
+    struct IMediaStream : virtual public Core::IUnknown {
 
-        enum { ID = ID_MEDIAPLAYER_NOTIFICATION };
+        enum { ID = ID_MEDIASTREAM };
 
-        virtual ~INotification() {}
+        struct INotification : virtual public Core::IUnknown {
 
-        virtual void Event(const string& eventName, const string& parametersJson) = 0;
-    };
+            enum { ID = ID_MEDIASTREAM_NOTIFICATION };
+
+            ~INotification() override = default;
+
+            virtual void Event(const string& eventName, const string& parametersJson) = 0;
+        };
 
-    virtual ~IMediaPlayer() {}
+        ~IMediaStream() override = default;
+
+        virtual uint32_t Load(const string& url, bool autoPlay) = 0;
+        virtual uint32_t SetRate(int32_t rate) = 0;
+        virtual uint32_t SeekTo(int32_t positionSec) = 0;
+        virtual uint32_t Stop() = 0;
+        virtual uint32_t InitConfig(const string& configurationJson) = 0;
```

Comment:
Do the `InitConfig` and `InitDRMConfig` methods intialize the player with a config? In which case, I'd call them `Configure` & `ConfigureDRM`, otherwise it might seem like you're initializing a configuration.

Reply:
Right, naming is confusing but adopted from UVE AAMP player: https://github.com/rdkcmf/rdk-aamp/blob/stable2/AAMP-UVE-API.pdf with the same functionality. Better to keep it as it is to not introduce additional confusion

Reply:
Good point, I haven't taken that into account.